raft/src/main/java/org/apache/kafka/raft/LogOffsetMetadata.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.raft;
+
+/**
+ * Metadata for specific local log offset
+ */
+public class LogOffsetMetadata {
+
+    public final long offset;
+    public final long segmentBaseOffset;
```

Comment:
It would be nice if we could keep the raft implementation somewhat agnostic to the lower level details of the log structure. My thought is that we could represent `LogOffsetMetadata` something like the following:

```java
class LogOffsetMetadata {
  final long offset;
  final Optional<OffsetMetadata> metadata
}

// Opaque metadata type
interface OffsetMetadata {}
```

Then inside `KafkaMetadataLog`, we can add a type which encapsulates the segment information and extends `OffsetMetadata`. What do you think?